doc/docs/Python_User_Interface.md

```diff
@@ -87,40 +87,40 @@ control various parameters of the Meep computation.
 
 ```python
 def __init__(self,
-             cell_size:Union[meep.geom.Vector3, Tuple[float, ...], NoneType]=None,
-             resolution:float=None,
-             geometry:Union[List[meep.geom.GeometricObject], NoneType]=None,
-             sources:Union[List[meep.source.Source], NoneType]=None,
-             eps_averaging:bool=True,
-             dimensions:int=3,
-             boundary_layers:Union[List[meep.simulation.PML], NoneType]=None,
-             symmetries:Union[List[meep.simulation.Symmetry], NoneType]=None,
-             force_complex_fields:bool=False,
-             default_material:meep.geom.Medium=Medium(),
-             m:float=0,
-             k_point:Union[meep.geom.Vector3, Tuple[float, ...], bool]=False,
-             kz_2d:str='complex',
-             extra_materials:Union[List[meep.geom.Medium], NoneType]=None,
-             material_function:Union[Callable[[Union[meep.geom.Vector3, Tuple[float, ...]]], meep.geom.Medium], NoneType]=None,
-             epsilon_func:Union[Callable[[Union[meep.geom.Vector3, Tuple[float, ...]]], float], NoneType]=None,
-             epsilon_input_file:str='',
-             progress_interval:float=4,
-             subpixel_tol:float=0.0001,
-             subpixel_maxeval:int=100000,
-             loop_tile_base_db:int=0,
-             loop_tile_base_eh:int=0,
-             ensure_periodicity:bool=True,
-             num_chunks:int=0,
-             Courant:float=0.5,
-             accurate_fields_near_cylorigin:bool=False,
-             filename_prefix:Union[str, NoneType]=None,
-             output_volume:Union[meep.simulation.Volume, NoneType]=None,
-             output_single_precision:bool=False,
-             geometry_center:Union[meep.geom.Vector3, Tuple[float, ...]]=Vector3<0.0, 0.0, 0.0>,
-             force_all_components:bool=False,
-             split_chunks_evenly:bool=True,
+             cell_size: Union[meep.geom.Vector3, Tuple[float, ...], NoneType] = None,
+             resolution: float = None,
+             geometry: Optional[List[meep.geom.GeometricObject]] = None,
+             sources: Optional[List[meep.source.Source]] = None,
```

Comment:
What's going on with all of these `Union -> Optional` changes?   Seems like they belong in a separate PR.

Reply:
#1919 added the `Optional` type hint to various parameters of the `Simulation` constructor such as `geometry`, `sources`, etc. The type hint `Optional[Type]` is equivalent to `Union[Type, None]` according to [PEP 484](https://peps.python.org/pep-0484/#union-types). The default values of each type are parsed by the `ast` module in [`doc/generate_py_api.py`](https://github.com/NanoComp/meep/blob/55f881fef388d591d201fec7b5b139d39db6747b/doc/generate_py_api.py#L140-L142). Because I have been using different versions of Python (3.5, 3.8) to call `generate_py_api.py` in various PRs which involve updating the docs (i.e., #1958, #1926, etc.), the different versions of `ast` seem to have been interchanging `Optional` and `Union`. The same thing seems to be happening again in this PR. `ast` is also inserting an extra space before and after the `=` sign. We therefore probably need to edit `generate_py_api.py` in a separate PR to ensure consistent parsing of the type hints and thus avoid these unnecessary changes.